src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -254,29 +351,44 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
             {
                 AsyncCompletionLogger.LogExpanderUsage();
 
+                // It's possible we didn't provide expanded items at the beginning of completion session because it was slow even if the feature is enabled.
+                // ExpandedItemsTask would be available in this case, so we just need to return its result.
+                // We are doing this in a spinning fashion because the lifetime of expandedItemsTask is associated with the completion session, we don't
+                // want to cancel that task when this "GetExpandedCompletionContextAsync" operation is being canceled, so we will still be able to retrieve
+                // the results later.
+                if (session.Properties.TryGetProperty<Task<(AsyncCompletionData.CompletionContext, CompletionList)>>(ExpandedItemsTask, out var expandedItemsTask))
```

Comment:
```suggestion
                if (session.Properties.TryGetProperty>(ExpandedItemsTask, out Task<(AsyncCompletionData.CompletionContext, CompletionList) expandedItemsTask))
```

Also, consider a using alias for this type (if you specify it in a few placed.